static/js/main.js

```diff
@@ -0,0 +1,168 @@
+ * @param {string} mode - The timer mode, formatted like so: `"<Work time>x<Rest time>"`
+ */
+function setMode(mode) {
+	currentWorkTime = mode.substring(0, mode.indexOf("x"));
+	currentRestTime = mode.substring(mode.indexOf("x") + 1);
+
+	buttons.forEach((b) => b.remove());
+
+	buttons = [
+		makeButton(
+			"start",
+			(currentButtonsAmount = 1),
+			() => (resetTimer = timerRunning = true)
+		),
+	];
+}
+
+/**
+ * Updates the screen to show the proper text about time left in a session.
+ * @param {number} workTime - The time to spend working
+ * @param {number} restTime - The time to spend resting
+ */
+function pomodoro(workTime, restTime) {
+	textSize((windowWidth + windowHeight) / 2 / 25);
+
+	if (resetTimer) {
+		buttons.forEach((b) => b.remove());
+		endTime = getTargetTime(working ? workTime : restTime);
+		buttons = [
+			makeButton(
+				"Reset",
+				(currentButtonsAmount = modes.length),
+				resetProgram
+			),
+		];
+
+		soundEffect.play();
+
+		resetTimer = false;
+	}
+
+	background(working ? color(0, 255, 0) : color(64, 64, 255));
+
+	timeLeft = getTimeDifference(getTime(), endTime);
+
+	if (timeLeft.seconds <= 0) {
+		working = !working;
+		resetTimer = true; //recalculate time to finish
+	}
+
+	const currentTimeInfo = new TimeInformation(
+		working,
+		workTime,
+		restTime,
+		timeLeft
+	);
+
+	theText = standardText(timeFormatter.textFrom(currentTimeInfo));
+}
+
+function standardText(txt) {
+	return text(txt, windowWidth / 2, windowHeight / 2);
+}
+
+/**
+ * Creates a list of buttons from a list of timer modes. Each button will start the corresponding mode when clicked.
+ * @param {string[]} modes - List of timer modes that the buttons should start.
+ * @returns {Button[]} The list of buttons that start the timer with their mode.
+ */
+function makeButtonsFromModes(modes) {
+	return modes
+		.map(createButton)
+		.map((b, idx) =>
+			b.position(
+				windowWidth / 2 - buttonWidth / 2,
+				windowHeight / 2 +
+					idx * buttonHeight -
+					(modes.length * buttonHeight) / 2
+			)
+		)
+		.map((b) => b.size(buttonWidth, buttonHeight))
+		.map((b, idx) => b.mousePressed(() => setMode(modes[idx])));
+}
+
+/**
+ * Creates a button on the screen.
+ * @param {string} txt - The text on the button
+ * @param {number} totalButtons - The intended total number of buttons
+ * @param {() => void} mPressed - The callback to be ran when the button is clicked
+ * @returns
+ */
+function makeButton(txt, totalButtons, mPressed) {
+	//currentButtonsAmount=the current number of buttons on screen before creation of this one
+	var button = createButton(txt);
+	button.position(
+		windowWidth / 2 - buttonWidth / 2,
+		windowHeight / 2 +
+			currentButtonsAmount * buttonHeight -
+			(totalButtons * buttonHeight) / 2
+	);
+	button.size(buttonWidth, buttonHeight);
+	button.mousePressed(mPressed);
+
+	currentButtonsAmount++;
+
+	return button;
+}
+
+window.setup = setup;
+window.mouseClicked = mouseClicked;
+window.draw = draw;
```

Comment:
Small trick for `p5.js` to "see" the functions we've declared when using the script with `type="module"`.